instrumentation/r2dbc-1.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/r2dbc/v1_0/R2dbcInstrumentationModule.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.instrumentation.r2dbc.v1_0;
+
+import com.google.auto.service.AutoService;
+import io.opentelemetry.javaagent.extension.instrumentation.InstrumentationModule;
+import io.opentelemetry.javaagent.extension.instrumentation.TypeInstrumentation;
+import java.util.Collections;
+import java.util.List;
+
+@AutoService(InstrumentationModule.class)
+public class R2dbcInstrumentationModule extends InstrumentationModule {
+
+  public R2dbcInstrumentationModule() {
+    super("r2dbc", "r2dbc-1.0");
+  }
+
+  @Override
+  public boolean isHelperClass(String className) {
+    return className.startsWith("io.r2dbc.proxy");
```

Comment:
So the alternative would be to do basically reimplement the `r2dbc-proxy` library in order to get Query Listeners just to not use `r2dbx-proxy` directly? It just seemed a bit redundant to me, but I get your point.
After @marcingrzejszczak  https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/7977#issuecomment-1456309254, should I be simply using the existing [ObservationProxyExecutionListener](https://github.com/r2dbc/r2dbc-proxy/blob/main/src/main/java/io/r2dbc/proxy/observation/ObservationProxyExecutionListener.java) or stick to the manually implemented `TraceProxyListener`, or even move away entirely from `r2dbc-proxy` for the auto instrumentation?

Reply:
We could shade the `r2dbc-proxy` library under a different package so it wouldn't conflict with whatever is in the application. I think this isn't super important, we can always fix it later. @marcingrzejszczak is one of the authors of micrometer tracing. As for his suggestion I'm not quite sure what he means. It could be either that he is suggesting to implement some sort of bridge with micrometer api so that their instrumentations could be automatically used or he might be hinting that instead of implementing this instrumentation for opentelemetry you could be using micrometer tracing where this instrumentation already exists.

Reply:
Yeah I'm suggesting that since the instrumentation is there providing a bridge would mean you have nothing to do :shrug: Otherwise you can just replicate the same code but use the OTel API.

Reply:
> We could shade the `r2dbc-proxy` library under a different package so it wouldn't conflict with whatever is in the application.

We're doing a similar thing in the [couchbase instrumentation](https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/instrumentation/couchbase/couchbase-3.2/javaagent/build.gradle.kts). @PhilKes you could probably implement it in a somewhat similar manner.

Reply:
@PhilKes I think it is better to to rename packages only in the code that gets embedded in the agent and leave the library module unshaded. I create a pr that applies on top of your pr for that https://github.com/PhilKes/opentelemetry-java-instrumentation/pull/1